File: selected_coding_interview/codes/python/lc_198_house_robber.py
"""
File: lc_198_house_robber.py
Created Time: 2025-12-30
Author: krahets
"""

from include import *


# ===== Solution Code =====
class Solution:
    def rob(self, nums: List[int]) -> int:
        cur, pre = 0, 0
        for num in nums:
            cur, pre = max(pre + num, cur), cur
        return cur


# ======= Test Case =======
# Test case 1
test_input = [1, 2, 3, 4, 5]

# ====== Driver Code ======
slt = Solution()
result = slt.rob(test_input)
print(result)

File: selected_coding_interview/codes/java/lc_198_house_robber/lc_198_house_robber.java
/*
* File: lc_198_house_robber.java
* Created Time: 2025-12-30
* Author: krahets
*/

package lc_198_house_robber;

import include.*;
import java.util.*;

// ===== Solution Code =====
    class Solution {
        public int rob(int[] nums) {
            int pre = 0, cur = 0, tmp;
            for(int num : nums) {
                tmp = cur;
                cur = Math.max(pre + num, cur);
                pre = tmp;
            }
            return cur;
        }
    }

public class lc_198_house_robber {
    public static void main(String[] args) {
        // ======= Test Case =======
        // Test case 1
        int[] test_input = new int[]{1, 2, 3, 4, 5};

        // ====== Driver Code ======
        Solution slt = new Solution();
        var result = slt.rob(test_input);
        System.out.println(result);

    }
}
